### PR TITLE
Bookstore: connect to API

### PR DESCRIPTION
**Pull request by Rudy-Menekam**
1. Make use of the [Bookstore API](https://www.notion.so/Bookstore-API-51ea269061f849118c65c0a53e88a739) as your backend
2. Refactored the code to use API calls to implement my functions(addBook and removeBook)

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -1,40 +1,64 @@
...
+
+export const removeBook = createAsyncThunk('removeBook', async (id) => {
+  await fetch(`${url}/${id}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  return id;
+});
```
Comment (binhussen) [inline src/redux/books/booksSlice.js:36]: - Great job for setuping redux :+1: however you are dispatching the remove action with an undefined parameter, so your delete functionality is not working both on the API method and redux-store filter :+1: please fix it to have a fully functioning bookstore :+1:

**Review by binhussen — CHANGES_REQUESTED** [review]
Hi @Rudy-Menekam  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights

- All linter checks passed ✔️
- Used GitH flow ✔️
- Used react-redux ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (Rudy-Menekam) [reply src/redux/books/booksSlice.js:36]: please @binhussen  i don't understand

**Review by MeqdamAlqudah — APPROVED** [review]
Hi @Rudy-Menekam 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
image

## Highlights 
:+1: no linter errors 
:+1: remove and add books

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**